oshi-core/src/main/java/oshi/driver/linux/proc/TaskStat.java

```diff
@@ -0,0 +1,259 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2010 - 2020 The OSHI Project Contributors: https://github.com/oshi/oshi/graphs/contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package oshi.driver.linux.proc;
+
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import oshi.software.common.AbstractOSThread;
+import oshi.software.os.OSProcess;
+import oshi.software.os.linux.LinuxOSProcess;
+import oshi.software.os.linux.LinuxOperatingSystem;
+import oshi.util.ExecutingCommand;
+import oshi.util.FileUtil;
+import oshi.util.ParseUtil;
+import oshi.util.platform.linux.ProcPath;
+
+public class TaskStat extends AbstractOSThread {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TaskStat.class);
+    private static final int PAGE_SIZE = ParseUtil
+            .parseIntOrDefault(ExecutingCommand.getFirstAnswer("getconf PAGESIZE"), 4096);
+
+    private static final int[] THREAD_PID_STAT_ORDERS = new int[TaskStat.ThreadPidStat.values().length];
```

Comment:
Minor comment on naming things (one of the two hard things).  The constant you're coping was intended to allude to `/proc/<pid>/stat` and you're still pulling stuff from the `/proc` filesystem.  So this should probably be `PROC_TASK_STAT_ORDERS` I think? `PROC_PID_TASK_STAT_ORDERS` is too verbose :)